app/controllers/api/proxy_rules_controller.rb

```diff
@@ -0,0 +1,67 @@
+class Api::ProxyRulesController < Api::BaseController
+  before_action :authorize_rolling_update, :find_service, :find_proxy
+
+  activate_menu :serviceadmin, :integration, :mapping_rules
+
+  sublayout 'api/service'
+
+  def index
+    @proxy_rules = @proxy.proxy_rules.includes(:metric).ordered.paginate(page: params[:page])
+  end
+
+  def new
+    last_position = @proxy.proxy_rules.maximum(:position) || 0
+    next_position = last_position + 1
+    @proxy_rule = @proxy.proxy_rules.build(position: next_position)
+  end
+
+  def create
+    @proxy_rule = @proxy.proxy_rules.build(proxy_rule_params)
+    if @proxy_rule.save
+      redirect_to admin_service_proxy_rules_path(@service), notice: 'Mapping rule was created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @proxy_rule = @proxy.proxy_rules.find(params[:id])
```

Comment:
`@proxy_rule = @proxy.proxy_rules.find(params[:id])` this should be extracted to a method and then have a `before_action :find_proxy_rule, only: %i[edit update destroy]`
and then you don't need to repeat this line at the beginning of the 3 methods 😄

Reply:
🥇

Reply:
@Martouta I did update the code with these suggestions